opm/test_util/summaryIntegrationTest.cpp

```diff
@@ -0,0 +1,417 @@
+/*
+  Copyright 2016 Statoil ASA.
+
+  This file is part of the Open Porous Media project (OPM).
+
+  OPM is free software: you can redistribute it and/or modify it under the terms of 
+  the GNU General Public License as published   by the Free Software Foundation, either 
+  version 3 of the License, or (at your option) any later version.
+
+  OPM is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with OPM.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+
+#include <opm/test_util/summaryIntegrationTest.hpp>
+#include <opm/common/ErrorMacros.hpp>
+#include <ert/ecl/ecl_sum.h>
+#include <ert/util/stringlist.h>
+
+
+void IntegrationTest::getIntegrationTest(){  
+    std::vector<double> timeVec1, timeVec2;
+    setTimeVecs(timeVec1, timeVec2);  // Sets the time vectors, they are equal for all keywords (WPOR:PROD01 etc)
+    setDataSets(timeVec1, timeVec2);
+
+    size_t ivar = 0;
+    if(!allowDifferentAmountOfKeywords){
+	if(stringlist_get_size(keysShort) != stringlist_get_size(keysLong)){
+	    OPM_THROW(std::invalid_argument, "Different ammont of keywords in the two summary files.");
+	}
+    }
+    if(printKeyword){printKeywords(); return;}
+
+  
+    std::string keywordWithGreatestErrorRatio;
+    double greatestRatio = 0;
+
+    //Iterates over all keywords from the restricted file, use iterator "ivar". Searches for a  match in the file with more keywords, use the itarator "jvar".
+    while(ivar < stringlist_get_size(keysShort)){
+	const char* keyword = stringlist_iget(keysShort, ivar);       
+	
+
+	if(oneOfTheMainVariables){
+	    std::string keywordString(keyword);
+	    std::string substr = keywordString.substr(0,4);
+	    if(substr!= mainVariable){
+		ivar++;
+		continue;
+	    }
+	}  
+    
+	for (size_t jvar = 0; jvar < stringlist_get_size(keysLong); jvar++){
+
+	    if (strcmp(keyword, stringlist_iget(keysLong, jvar)) == 0){ //When the keywords are equal, proceed in comparing summary files. 
+	
+		//	if(!checkUnits(keyword)){OPM_THROW(std::runtime_error, "For keyword " << keyword << " the unit of the two files is not equal. Not possible to compare.");} //Comparing the unit of the two vectors. 	
```

Comment:
The reason why I have commented this code out is that flow does not write out which unit the keyword vector has. I was thinking this could be commented until flow writes out units, and than be uncommented. What do you think?